app/public/js/localSprites.js

```diff
@@ -254,8 +250,8 @@ function addFireballParticle(parent, decayFrames, parentPreScalingXSize, parentP
         randomY = Math.round(parent.y - ((Math.random() * parent.yScale * parentPreScalingYSize) / 2));
     }
     var fireballParticle = new SimpleSprite({frames}, randomX, randomY, 0, 0, 1.25, 2.5);
-    if (type === 0) fireballParticle.name = 'fireballContrailParticle';
-    if (type === 1) fireballParticle.name = 'fireballDetonationParticle';
+    if (type === 0) fireballParticle.type = 'fireballContrailParticle';
```

Comment:
Once you pass in the correct `SPRITE_TYPE_X` constant here, you can remove these if statements and repplace this line with `fireballParticle.type = type`